src/test/java/com/artipie/rpm/pkg/PrimaryOutputTest.java

```
@@ -0,0 +1,115 @@
+class PrimaryOutputTest {
+
+    @Test
+    void createsPrimaryForAbc(@TempDir final Path temp) throws IOException {
+        final Path res = temp.resolve("primary.xml");
+        try (PackageOutput.FileOutput primary = new PrimaryOutput(res)) {
+            primary.start();
+            final Path rpm =
+                Paths.get("src/test/resources-binary/abc-1.01-26.git20200127.fc32.ppc64le.rpm");
+            primary.accept(new FilePackage.Headers(this.header(rpm), rpm, Digest.SHA256));
+        }
+        MatcherAssert.assertThat(
+            new String(Files.readAllBytes(res), StandardCharsets.UTF_8),
+            CompareMatcher.isIdenticalTo(
+                new String(
+                    Files.readAllBytes(
+                        Paths.get("src/test/resources-binary/repodata/abc-primary.xml")
+                    ),
+                    StandardCharsets.UTF_8
+                )
+            ).ignoreWhitespace()
+            .ignoreElementContentWhitespace()
+            .normalizeWhitespace()
+            .withNodeFilter(
+                node -> !"file".equals(node.getLocalName())
+                    && !"provides".equals(node.getLocalName())
+                    && !"requires".equals(node.getLocalName())
+            ).withAttributeFilter(
+                attr -> !"file".equals(attr.getName()) && !"archive".equals(attr.getName())
+            )
+        );
+    }
+
+    /**
+     * Get header.
+     * @return The header
+     * @param file Rpm file path
+     * @throws InvalidPackageException In case package is invalid.
+     * @throws IOException On error
+     * @todo #241:30min Introduce class from this method: the method is copied from FilePackage
+     *  class, we need to extract class from it, test it and use it at least in FilePackage,
+     *  ModifiableMetadataTest and here.
+     */
+    @SuppressWarnings("PMD.AvoidCatchingGenericException")
+    private Header header(final Path file) throws InvalidPackageException, IOException {
```

@olenagerasimova `InvalidPackageException` doesn't need to be declared explicitly. Also I don't think that is a good idea to throw both runtime and catch exceptions. I would throw only `InvalidPackageException` or only `IOException`. wdyt?